src/app/modules/form/admin-form/admin-form.controller.ts

```diff
@@ -230,3 +231,54 @@ export const handleCountFormSubmissions: RequestHandler<
   // Successfully retrieved count.
   return res.json(countResult.value)
 }
+
+/**
+ * Handler for GET /{formId}/adminform/feedback/count.
+ * @security session
+ *
+ * @returns 200 with feedback counts of given form
+ * @returns 403 when user does not have permissions to access form
+ * @returns 404 when form cannot be found
+ * @returns 410 when form is archived
+ * @returns 422 when user in session cannot be retrieved from the database
+ * @returns 500 when database error occurs
+ */
+export const handleCountFormFeedback: RequestHandler<{
+  formId: string
+}> = async (req, res) => {
+  const { formId } = req.params
+  const sessionUserId = (req.session as Express.AuthedSession).user._id
+
+  return (
+    // Step 1: Retrieve currently logged in user.
+    UserService.getPopulatedUserById(sessionUserId)
+      .andThen((user) =>
+        // Step 2: Retrieve full form.
+        FormService.retrieveFullFormById(formId).andThen((fullForm) =>
+          // Step 3: Check whether form is active.
+          assertFormAvailable(fullForm).andThen(() =>
+            // Step 4: Check whether current user has read permissions to form.
+            assertHasReadPermissions(user, fullForm),
```

Comment:
lol the nesting makes me sad but idk any other way to do it without doing it one step at a time